wasm/validation.h

```diff
@@ -0,0 +1,115 @@
+// SPDX-FileCopyrightText: 2024 David Zero <zero-one@zer0-one.net>
+//
+// SPDX-License-Identifier: BSD-2-Clause
+
+#ifndef WASM_VALIDATION_H_
+#define WASM_VALIDATION_H_
+
+#include "wasm.h"
+#include "wasm/instructions.h"
+#include "wasm/types.h"
+
+#include <tl/expected.hpp>
+
+#include <cstdint>
+#include <string_view>
+#include <variant>
+#include <vector>
+
+namespace wasm::validation {
+
+// https://webassembly.github.io/spec/core/valid/types.html#limits
+constexpr bool is_valid(Limits const &l, std::uint64_t k) {
```

Comment:
If these are only used internally in the validation, then consider moving them into the .cpp (they can still be constexpr etc.)